vendor/bat-native-ledger/src/bat/ledger/internal/promotion.cc

```diff
@@ -40,6 +40,34 @@ using challenge_bypass_ristretto::UnblindedToken;
 
 namespace braveledger_promotion {
 
+namespace {
+
+void HandleExpiredPromotions(
+    bat_ledger::LedgerImpl* ledger,
+    ledger::PromotionMap* promotions) {
+  // TODO(https://github.com/brave/brave-browser/issues/6363): Filtering logic
```

Comment:
not sure about this TODO. It links to move db into ledger. How are they related

Reply:
On speaking with @tmancey regarding refactoring the database, this kind of operation should be a query executed within the DB, not in code. We could actually change to this now, but I've no oversight on the DB refactor at the minute. What do you think?

Reply:
personally I would remove this TODO and a reference to the issue as it will not fix in that issue

Reply:
Done